### PR TITLE
Feat/post and tag dto

### PR DESCRIPTION
**Pull request by Mimachh**
Tu as un peu de lecture, désolé.

### CODE REVIEW

File: src/post/post.service.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                create: postTypeChoiceCreate,
+            },
+          },
+        });
+    }
+
+    async editPost(postId: number, updatePostDto: UpdatePostDto) {
+        const { description, userId, tags, postBody } = updatePostDto;
+
+        // I tried this to check if the user can update. Is it good ?
```
Comment (antikkorps) [inline src/post/post.service.ts:102]: Here you can add a where constraint in order to keep the code clean. What you've done is certainly working but and this add a condition where a single constraint in Prisma should be fine

File: types/all.ts
Comment (antikkorps) [inline types/all.ts:1]: This is really cool like that

File: src/post/post.service.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                },
+                postTypeChoice: {
+                    create: postTypeChoiceCreate,
+                },
+            }
+        })
+
+        return updatedPost;
+    }
+    // I list the method we will probably need
```
Comment (antikkorps) [inline src/post/post.service.ts:150]: I am working on that kind of stuff on the api of myEvaluation, there is one thing that can be donne is creating a route like /post/user/:id where the id is the user id. As though we can have a specific route to the method. What you think about it?

File: NewRemarques.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Il faut réfléchir aux story
+Les commentaires doivent acceuillir des réponses et des réactions.
```
Comment (antikkorps) [inline NewRemarques.md:2]: Yes right. There we can use the polymorphic table to likes and comments and reactions. All the three

File: NewRemarques.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Il faut réfléchir aux story
```
Comment (antikkorps) [inline NewRemarques.md:1]: There we have to make difference from insta I think. Perhaps all post from the day should go in story. What do you think?

**Review by antikkorps — CHANGES_REQUESTED** [review]

File: src/post/post.service.ts (hunk 2 above)
Comment (Mimachh) [reply src/post/post.service.ts:150]: A route to the user who's update the post ?
Yes we can do that, and create a most global fonction maybe to check if user is owner or admin ?

Comment (Mimachh) [reply src/post/post.service.ts:150]: Yes that was my first idea, then I thought about the findAll with some specific params to get the post's user.
But having a specific route is far better. 
/post/user/:id is good. 
Or maybe user/:id/post  <- ? Then if we had params it's more readable : 
- user/1/post?tags=xxxx&limit=3
- or post/user/1?tags=xxx&limit=3
What's better for you?

File: NewRemarques.md (hunk 3 above)
Comment (Mimachh) [reply NewRemarques.md:2]: You think we can use the same polymorphic table for all the three?

File: NewRemarques.md (hunk 4 above)
Comment (Mimachh) [reply NewRemarques.md:1]: Why not.
Do we have pined post? Then user can pin a specific post to his profile?

File: src/post/post.service.ts (hunk 2 above)
Comment (antikkorps) [reply src/post/post.service.ts:150]: it won't be readable for both as the user_id is a uuid. In my opinion post/user/1?tags=xxx&limit=3 would be better

File: NewRemarques.md (hunk 3 above)
Comment (antikkorps) [reply NewRemarques.md:2]: I think this would work for the three yes

File: NewRemarques.md (hunk 4 above)
Comment (antikkorps) [reply NewRemarques.md:1]: just to differentiate the service, make something a little bit different but there is no other reason. We don't have actually pined post but we could and perhaps should in some way in order to give them the ability to show something very specific on top of their profile.

File: src/post/post.service.ts (hunk 2 above)
Comment (Mimachh) [reply src/post/post.service.ts:150]: Yes I add it in the future push. 
Anyway the user/id/post was more painful to create cause the default route in the post folder was /post/* 
It's simplier to keep the /post/user/id

**Review by Mimachh — COMMENTED** [review]
Voili voilou, je finish pour pouvoir repush derrière. On va pas se mentir je le fais que par intérêt. Comme une vulgaire cagole.

**Pull request merged**